wellcomeml/datasets/winer.py

```diff
@@ -0,0 +1,271 @@
+
+    Utilise the three datasources to get the entities from a sample
+    of documents and save to a training and testing dataset files
+    in the form:
+
+    ID 12345
+
+    John Smith PERSON
+    rejects O
+    German LOC
+    call O
+    
+    Next 0
+    sentence 0
+
+    ID 444
+
+    Next
+    document
+    """
+
+    # Set the random seed for taking a sample
+    random.seed(rand_seed)
+    n_train = round(prop_train*n_sample)
+
+    # Make a dictionary for the entities for every article
+    # in the form
+    # entities = {article_id: {sentence_id:[[0, 4, 0], [18, 25, 1]]}}
+    # The sentence id is which number sentence (e.g. 1st, 2nd) the entities
+    # are from in the whole document
+
+    logger.info("Creating entities dictionary from {}".format(NE_path))
+    entities = {}
+    with tarfile.open(NE_path, "r:bz2") as tar:
+        for member in tqdm(tar.getmembers()):
+            f = tar.extractfile(member)
+            if f:
+                content = f.read().decode('utf-8', errors='ignore')
+                articles = content.replace('\t', ' ').split('ID ')
+                # The first element will be blank from the splitting
+                articles = articles[1:]
+                for article in articles:
+                    article_entities = article.split('\n')
+                    article_id = article_entities[0]    
+                    # The first and possibly last entity will be blank from the splitting
+                    if article_entities[-1]=='':
+                        article_entities = article_entities[1:-2]
+                    else:
+                        article_entities = article_entities[1:]
+                    entity_info = [l.split(' ') for l in article_entities]
+                    sentence_entities = {}
+                    for sentence_id, begin, end, ent_type in entity_info:
+                        # The data is stored as strings, but they are all numerical
+                        # and sentence id = 3 refers to the 2nd sentence in this article
+                        # so for querying the list of sentences in an article
+                        # later it is useful to have this as an integer
+                        sentence_id, begin, end, ent_type = int(sentence_id), int(begin), int(end), int(ent_type)
+                        if sentence_entities.get(sentence_id):
+                            sentence_entities[sentence_id].append([begin, end, ent_type])
+                        else:
+                            sentence_entities[sentence_id] = [[begin, end, ent_type]]
+                    entities[article_id] = sentence_entities
+
+    # Create id2word from vocab file
+    # The WiNER documentation states that the line number is the word id
+    logger.info("Creating id2word dictionary from {}".format(vocab_path))
+    id2word = {}
+    with open(vocab_path, "r") as vocab:
+        f = vocab.read()
+        lines = f.split('\n')
+        for i, line in enumerate(lines):
```

Comment:
i think you can just do `or i, line in enumerate(vocab):` and remove the above two lines

Reply:
oh cool, didnt think that